core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/machineLearning/sklearnAdvanced/base/SklearnAdvancedBaseDesc.scala

```diff
@@ -0,0 +1,154 @@
+        str1 = str1 + String.format("%s = {},", ele.parameter.getName)
+        str2 =
+          str2 + String.format("%s(table['%s'].values[i]),", ele.parameter.getType, ele.attribute)
+        str3 = str3 + String.format(
+          "%s = %s(table['%s'].values[i]),",
+          ele.parameter.getName,
+          ele.parameter.getType,
+          ele.attribute
+        )
+      } else {
+        str1 = str1 + String.format("%s = {},", ele.parameter.getName)
+        str2 = str2 + String.format("%s ('%s'),", ele.parameter.getType, ele.value)
+        str3 = str3 + String.format(
+          "%s = %s ('%s'),",
+          ele.parameter.getName,
+          ele.parameter.getType,
+          ele.value
+        )
+      }
+    }
+    List(String.format("\"%s\".format(%s)", str1, str2), str3)
+  }
+
+  override def generatePythonCode(): String = {
+    val listFeatures = selectedFeatures.map(feature => s""""$feature"""").mkString(",")
+    val trainingName = getImportStatements.split(" ").last
+    val stringList = getParameter(paraList)
+    val trainingParam = stringList(1)
+    val paramString = stringList(0)
+    val finalCode =
+      s"""
+         |from pytexera import *
+         |
+         |import pandas as pd
+         |${getImportStatements}
+         |
+         |class ProcessTableOperator(UDFTableOperator):
+         |
+         |  @overrides
+         |  def process_table(self, table: Table, port: int) -> Iterator[Optional[TableLike]]:
+         |    global dataset
+         |    model_list = []
+         |    para_list = []
+         |    features = [$listFeatures]
+         |
+         |    if port == 0:
+         |      dataset = table
+         |
+         |    if port == 1 :
+         |      y_train = dataset["$groundTruthAttribute"]
+         |      X_train = dataset[features]
+         |      loop_times = ${getLoopTimes(paraList)}
+         |
+         |      for i in range(loop_times):
+         |        model = ${trainingName}(${trainingParam})
+         |        model.fit(X_train, y_train)
+         |        model_list.append(model)
+         |        para_str = ${paramString}
+         |        para_list.append(para_str)
+         |
+         |      data = dict({})
+         |      data["Model"]= model_list
+         |      data["Parameters"] =para_list
+         |
+         |      df = pd.DataFrame(data)
+         |      yield df
+         |
+         |""".stripMargin
+    finalCode
+  }
+
+  override def operatorInfo: OperatorInfo = {
+    val name = getOperatorInfo
+    OperatorInfo(
+      name,
+      "Sklearn " + name + " Operator",
+      OperatorGroupConstants.ADVANCED_SKLEARN_GROUP,
+      inputPorts = List(
+        InputPort(
+          PortIdentity(0),
+          displayName = "dataset",
+          allowMultiLinks = true
+        ),
+        InputPort(
+          PortIdentity(1),
+          displayName = "parameter",
+          allowMultiLinks = true,
+          dependencies = List(PortIdentity(0))
```

Comment:
does this has to be a port? in the first version, can we just use the parameters from the property panel? 

If you really need to include this port, then questions:
1. why does it allow multiple links? could you have multiple upstreams?
2. why does it depend on port 0? In my opinion, parameters can come first then training dataset can come second.

Reply:
1. Fixed
2. We have to get the dataset from port 0, then get the parameters from port 1 and train the data. We discuss in detail with Xinyuan this afternoon.